src/lib.rs

```diff
@@ -133,23 +135,24 @@ where
 {
     type Item = (&'a Key, &'a Value);
 
-    #[cfg_attr(feature = "cargo-clippy", allow(while_let_on_iterator))]
     fn next(&mut self) -> Option<(&'a Key, &'a Value)> {
-        while let Some((key, &(ref value, _))) = self.map_iter.next() {
-            if !self.lru_cache.expired(key) {
-                return Some((key, value));
-            }
-        }
-        None
+        let now = Instant::now();
+        let not_expired = match self.lru_cache_ttl {
+            Some(ttl) => self
+                .map_iter
+                .find(|&(_, &(_, instant))| instant + ttl > now),
+            None => self.map_iter.next(),
+        };
+        not_expired.map(|(key, &(ref value, _))| (key, value))
     }
 }
 
 /// Implementation of [LRU cache](index.html#least-recently-used-lru-cache).
 pub struct LruCache<Key, Value> {
     map: BTreeMap<Key, (Value, Instant)>,
-    list: VecDeque<Key>,
+    list: Vec<Key>,
```

Comment:
Since we `pop_front()` (or rather `drain(..n)` from the front) a lot, wouldn't a `VecDeque` still be more efficient?

Reply:
Probably ... this is the part I am not sure about.

In a perfect world, I think we should probably remove the the expired element only after a certain threshold is reached. Hard to explain but something like the `Vec::capacity` mechanism.